src/api/handlers/appointments/sendReminders.js

```diff
@@ -0,0 +1,80 @@
+import moment from "moment";
+
+import { db } from "../../../../knex";
+import { sendMessage } from "../../../services/twilioClient";
+import TemplatesModel from "../../../models/templates";
+import LanguagesModel from "../../../models/languages";
+
+const daysFromNow = (interval) => {
+  return moment().add(interval, "d").format("YYYY-MM-DD hh:mm:ss");
+};
+
+export default async (req, res) => {
+  let appointments;
+  try {
+    appointments = await db("appointments")
+      .select("*")
+      .where("appointmentTime", ">=", daysFromNow(1))
+      .where("appointmentTime", "<", daysFromNow(3));
+  } catch (error) {
+    res.status(500).send({
+      error,
+      message: `Could not retrieve appointments for ${daysFromNow(1)}`
+    });
+  }
+
+  const appointmentsPromises = appointments.map(appointment => {
+    const {
+      appointmentId,
+      patientLanguage,
+      patientPhoneNumber
+    } = appointment;
+  
+    const language = LanguagesModel.getByLanguageString(patientLanguage);
+    // 1 is reminder template ID, full templated needed for name in message record
+    const template = TemplatesModel.getById(1);
+    const messageBody = TemplatesModel.generateMessage(1, language.id, appointment);
+
+    const message = {
+      appointmentId,
+      messageBody,
+      language: language.id,
+      templateName: template.templateName,
+      // UTC, will be the same for every message sent in a batch
+      timeSent: moment().format("YYYY-MM-DD HH:mm:ss")
+    };
+
+    return new Promise((resolve) => {
```

Comment:
I think this would work too and you handle the database error in the catch or append a .catch to the db call... But I am just wondering, do you need that `new Promise` there?
```
    return sendMessage(patientPhoneNumber, messageBody)
      .then(() => db("messages").insert(message))
      .then((message) => ({
          appointmentId,
          messageId: messageId[0]
      }))
      // TODO: don't resolve this, create a log or send a notification to someone
      .catch((error) => {
        // Handle DB error here too
        return {
          error,
          appointmentId
        };
      });
```

Reply:
@iamstevedavis I think I do since I want to handle each individual sendMessage then/catch rather than use the Promise.all then/catch (since sendMessage are each a self-contained unit and I don't want one failure to result in a general failure), but I'm not sure.

Reply:
Ahhhh I see what you mean, kind of like this:
```

const arrayOfPromises = [
  Promise.reject("Rejected").catch(() => Promise.resolve("Rejected")),
  Promise.resolve("Resolve"),
  Promise.resolve("Resolve"),
  Promise.reject("Rejected").catch(() => Promise.resolve("Rejected"))
];

return Promise.all(arrayOfPromises)
  .then((results) => {
    console.log(results);
  });
```
Well then :man_shrugging: looks good I guess aha

Reply:
Yeah, exactly. I could probably make it more obvious what I'm trying to do.